### PR TITLE
Add batched driver test for periodic LCAO.

### LINKED ISSUE #4782: batched VMC/DMC single determinant + LCAO tests needed
I cannot find one but I can be wrong.
determinisitic/short/long?

### PR DESCRIPTION
**Pull request by Hyeondeok-Shin**
## Proposed changes

Close #4782.
This PR is to add batched driver test for periodic LCAO diamond 1x1x1 and 2x1x1.

## What type(s) of changes does this code introduce?

- Other (please describe): Add test

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
Workstation

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: tests/solids/diamondC_2x1x1-Gaussian_pp/CMakeLists.txt [hunk 1]
```diff
@@ -87,3 +123,15 @@ qmc_run_and_check(
   0
   DET_DIAMOND_SCALARS # VMC
 )
+
+qmc_run_and_check(
+  deterministic-diamondC_2x1x1_pp-vmcbatch_gaussian_sdj
+  "${qmcpack_SOURCE_DIR}/tests/solids/diamondC_2x1x1-Gaussian_pp"
```
Comment (ye-luo) [inline tests/solids/diamondC_2x1x1-Gaussian_pp/CMakeLists.txt:129]: Can we also have vmcbatch_dmcbatch with 1 MPI 4 threads,  2 MPI 2 threads,  2 MPI 3 threads variants?

File: tests/solids/diamondC_2x1x1-Gaussian_pp/CMakeLists.txt [hunk 2]
```diff
@@ -21,6 +21,18 @@ qmc_run_and_check(
   DIAMOND_SCALARS # VMC
 )
 
+qmc_run_and_check(
```
Comment (ye-luo) [inline tests/solids/diamondC_2x1x1-Gaussian_pp/CMakeLists.txt:24]: Please remove the above swap file.

**Comment by ye-luo** [comment]
Test this please

File: tests/solids/diamondC_1x1x1-Gaussian_pp/CMakeLists.txt [hunk 3]
```diff
@@ -125,11 +125,11 @@ qmc_run_and_check(
...
   list(APPEND DET_DIAMOND_VMCBATCH_SCALARS "totenergy" "-10.82674512 0.000222")
   list(APPEND DET_DIAMOND_VMCBATCH_SCALARS "kinetic" "12.23702380 0.000221")
-  list(APPEND DET_DIAMOND_VMCBATCH_SCALARS "potential" "-23.06376891 0.000005")
+  list(APPEND DET_DIAMOND_VMCBATCH_SCALARS "potential" "-23.06376891 0.00003")
   list(APPEND DET_DIAMOND_VMCBATCH_SCALARS "eeenergy" "-3.00267948 0.000003")
   list(APPEND DET_DIAMOND_VMCBATCH_SCALARS "ionion" "-12.77567012 0.000001")
-  list(APPEND DET_DIAMOND_VMCBATCH_SCALARS "localecp" "-7.99462588 0.000007")
-  list(APPEND DET_DIAMOND_VMCBATCH_SCALARS "nonlocalecp" "0.70920650 0.000003")
+  list(APPEND DET_DIAMOND_VMCBATCH_SCALARS "localecp" "-7.99462588 0.00003")
+  list(APPEND DET_DIAMOND_VMCBATCH_SCALARS "nonlocalecp" "0.70920650 0.00001")
```
Comment (ye-luo) [inline tests/solids/diamondC_1x1x1-Gaussian_pp/CMakeLists.txt:132]: In the failed result checking of the last run, I saw all the values with 2X of the error bars before your changes. Can we have more conservative choice of 3X? Or you have a particular reason having up-to 6X?

Comment (Hyeondeok-Shin) [reply tests/solids/diamondC_1x1x1-Gaussian_pp/CMakeLists.txt:132]: I just increase as much as the deviation from the reference for the safety. I can adjust the error to fit the tolerance not the deviation.

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Pull request merged**